modules/extensions-support/src/main/java/org/mule/module/extension/internal/runtime/config/DynamicConfigurationInstanceHolder.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.runtime.config;
+
+import org.mule.extension.runtime.Expirable;
+import org.mule.extension.runtime.ExpirationPolicy;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * A holder class which decorates a {@link #configurationInstance} with extra
+ * management information such as registration name, usage tracking, etc.
+ * <p/>
+ * This class is thread safe.
+ *
+ * @since 4.0
+ */
+final class DynamicConfigurationInstanceHolder implements Expirable
```

Comment:
Is not about referencing every implemented interface. Saying that something is a "holder" is pretty vague.